scripts/create-associations.js

```diff
@@ -28,65 +28,117 @@ var credentials = {
+var progressBar;
 
 console.log(process.argv.slice(2).join(' '));
 
-Bluebird.resolve(dfp.getLineItems(all))
-  .then(function(lineItems) {
-    console.log('got all line items');
-    return [lineItems, dfp.getCreatives(all)];
-  })
-  .spread(function(lineItems, creatives) {
-    var associations = {};
-    console.log('got all creatives');
-
-    lineItems.forEach(function(lineItem) {
-      associations[lineItem.name] = {
-        lineItemId: lineItem.id
-      };
-    });
-
-    creatives.forEach(function(creative) {
-      if (associations[creative.name]) {
-        associations[creative.name].creativeId = creative.id;
-      }
-    });
-
-    return associations;
-  })
-  .then(function(associations) {
-    var ids = _.map(associations, function(associationIds, names) {
-      return associationIds;
-    });
-    ids = _.compact(ids);
-    return dfp.createAssociations(ids);
-  })
-  .then(function(associations) {
+function prepareQuery(){
+  var allLineItems = [
+    channel,
+    platform + size + position,
+    region,
+    partner,
+    WILDCARD
+  ].join('_').toUpperCase();
+
+  return allLineItems;
+}
+
+function getLineItems(query){
+  return [dfp.getLineItems(query), query];
+}
+
+function getCreatives(lineItems, query) {
+  console.log('got all line items');
+  return [lineItems, dfp.getCreatives(query)];
+}
+
+function combineByName(lineItems, creatives) {
+  var associations = {};
+  console.log('got all creatives');
+
+  lineItems.forEach(function(lineItem) {
+    associations[lineItem.name] = {
+      lineItemId: lineItem.id
+    };
+  });
+
+  creatives.forEach(function(creative) {
+    if (associations[creative.name]) {
+      associations[creative.name].creativeId = creative.id;
+    }
+  });
+
+  return associations;
+}
+
+function prepareAssociations(ids) {
+  var associations = _.map(ids, function(associationIds, names) {
+    return associationIds;
+  });
+  associations = _.compact(associations);
+  return associations;
+}
+
+function createAssociations(ids){
+  return dfp.createAssociations(ids)
+    .tap(advanceProgress);
+}
+
+function logSuccess(results) {
+  advanceProgress();
+  if (results){
     console.log('created associations');
-  })
-  .catch(function(err) {
-    console.log('creating all associations failed');
-    console.log('because', err.stack);
+  }
+}
+
+function handleError(err){
+  console.log('creating all associations failed');
+  console.log('because', err.stack);
+}
+
+function splitBatches(lineItems){
+  var batches = _.chunk(lineItems, 400);
+  progressBar = new ProgressBar('Progress [:bar] :percent :elapseds', {
+    total: batches.length + 1
   });
+  return batches;
+}
+
+function advanceProgress(){
+  progressBar.tick();
+}
+
+// this function is to help debugging
+function log(x){
+  console.log(x);
+}
+
+Bluebird.resolve(prepareQuery())
+  .then(getLineItems)
+  .spread(getCreatives)
+  .spread(combineByName)
+  .then(prepareAssociations)
+  .then(splitBatches)
```

Comment:
I added the pattern of batching to all scripts that could possibly create more than around 400 items. If you create less than 400, they'll just execute one batch and basically work the same way, although this will give us a nice progress bar.